examples/spark-history-server.yaml

```diff
@@ -0,0 +1,42 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: spark.apache.org/v1beta1
+kind: SparkApplication
+metadata:
+  name: spark-history-server
+spec:
+  mainClass: "org.apache.spark.deploy.history.HistoryServer"
+  sparkConf:
+    spark.jars.packages: "org.apache.hadoop:hadoop-aws:3.4.1"
+    spark.jars.ivy: "/tmp/.ivy2.5.2"
+    spark.driver.memory: "2g"
+    spark.kubernetes.authenticate.driver.serviceAccountName: "spark"
+    spark.kubernetes.container.image: "apache/spark:4.0.0-java21-scala"
+    spark.history.fs.logDirectory: "s3a://spark-events"
+    spark.history.fs.cleaner.enabled: "true"
+    spark.history.fs.cleaner.maxAge: "30d"
+    spark.history.fs.cleaner.maxNum: "100"
+    spark.history.fs.eventLog.rolling.maxFilesToRetain: "10"
+    spark.hadoop.fs.defaultFS: "s3a://spark-events"
+    spark.hadoop.fs.s3a.endpoint: "http://localstack:4566"
+    spark.hadoop.fs.s3a.path.style.access: "true"
+    spark.hadoop.fs.s3a.access.key: "test"
+    spark.hadoop.fs.s3a.secret.key: "test"
+  runtimeVersions:
+    sparkVersion: "4.0.0"
+  applicationTolerations:
+    restartConfig:
+      restartPolicy: Always
+      maxRestartAttempts: 9223372036854775807
```

Comment:
Since the default value of `maxRestartAttempts` is 3 (for normal jobs), I set this as `Long.MAX_VALUE` because SHS is supposed to run forever.
```bash
$ jshell
|  Welcome to JShell -- Version 21.0.7
|  For an introduction type: /help intro

jshell> Long.MAX_VALUE
$1 ==> 9223372036854775807
```